libs/designsystem/src/lib/components/dropdown/dropdown.component.ts

```diff
@@ -268,28 +264,44 @@ export class DropdownComponent
     }
   }
 
-  private setCardAlignment(entry) {
+  private setHorizontalDirection(entry) {
     // If card alignment is left, and the entry is cut off to the right by ${entry.boundingClientRect.right - entry.intersectionRect.right}px
     // it is set to align to end in stead, and vice versa for right-aligned card
-    if (this.cardAlign === cardAlignment.left) {
+    if (this._horizontal === HorizontalDirection.right) {
       if (entry.boundingClientRect.right > entry.rootBounds.right) {
-        this._horizontal = 'end';
-        this.cardAlign = cardAlignment.right;
+        this._horizontal = HorizontalDirection.left;
       }
     } else {
       if (entry.boundingClientRect.left < entry.rootBounds.left) {
-        this._horizontal = 'start';
-        this.cardAlign = cardAlignment.left;
+        this._horizontal = HorizontalDirection.right;
+      }
+    }
+  }
+
+  private setVerticalDirection(entry) {
+    if (entry.boundingClientRect.top < 0) {
+      // entry is cut off at the top by ${entry.boundingClientRect.top}px
+      // open downwards:
+      this._vertical = VerticalDirection.down;
+    }
+    if (entry.boundingClientRect.bottom > entry.rootBounds.bottom) {
+      // entry is cut off at the bottom by ${entry.boundingClientRect.bottom - entry.intersectionRect.bottom}px
+      const containerOffsetTop = this.elementRef.nativeElement.getBoundingClientRect().top;
+      const SPACING = 5; //TODO: Get from SCSS
+      // Check if the card can fit on top of button:
+      if (containerOffsetTop > entry.target.clientHeight + SPACING) {
+        // open upwards:
+        this._vertical = VerticalDirection.up;
       }
     }
   }
 
-  open() {
+  openDropdown() {
```

Comment:
Unfortunately `open()` is part of the public and published API, so shouldn't change (or at least have a very good reason to...)
 I think we'll have to keep it and rename the direction input to `openDirection` or similar...